bam-index/src/main/java/org/victorchang/KeyPointerReader.java

```diff
@@ -36,22 +36,22 @@
             /**
              * reusable buffer to minimize memory allocation.
              */
-            private final byte[] inputBuff = new byte[256 + 8 + 2];
+            private final byte[] inputBuff = new byte[256];
             private KeyPointer current = null;
 
             @Override
             public boolean hasNext() {
                 if (current == null) {
                     try {
-                        int entryLen;
+                        int keyLen;
                         try {
-                            entryLen = dataInput.readShort();
+                            keyLen = dataInput.readByte();
```

Comment:
you need `keyLen = dataInput.readUnsignedByte()` otherwise it may return negative number, byte is signed [-127,127]

Reply:
Good catch! I'll add a test as well.

Reply:
Fix: https://github.com/VCCRI/atlantool/pull/24